event-processing/event-processor/tests/test-events/unknown-audit-event.ts

```diff
@@ -1,96 +1,96 @@
-        message.timestamp !== undefined && (obj.timestamp = Math.round(message.timestamp));
-        message.timestamp_formatted !== undefined && (obj.timestamp_formatted = message.timestamp_formatted);
-        message.event_name !== undefined && (obj.event_name = message.event_name);
-        message.component_id !== undefined && (obj.component_id = message.component_id);
-        message.user !== undefined &&
-            (obj.user = message.user ? AuditEventUserMessage.toJSON(message.user) : undefined);
-        message.platform !== undefined &&
-            (obj.platform = message.platform ? message.platform : undefined);
-        message.restricted !== undefined &&
-            (obj.restricted = message.restricted ? message.restricted : undefined);
-        message.extensions !== undefined &&
-            (obj.extensions = message.extensions ? message.extensions : undefined);
-        message.persistent_session_id !== undefined && (obj.persistent_session_id = message.persistent_session_id);
-        message.new_unknown_field !== undefined && (obj.new_unknown_field = message.new_unknown_field);
-        return obj;
-    }
-}
-
-export class AuditEventUserMessage {
-    static fromJSON(object: any): IAuditEventUserMessage {
-        return {
-            transaction_id: isSet(object.transaction_id) ? String(object.transaction_id) : '',
-            email: isSet(object.email) ? String(object.email) : '',
-            phone: isSet(object.phone) ? String(object.phone) : '',
-            ip_address: isSet(object.ip_address) ? String(object.ip_address) : '',
-            unknown_user_field: isSet(object.unknown_user_field) ? String(object.unknown_user_field) : '',
-        };
-    }
-
-    static toJSON(message: IAuditEventUserMessage): unknown {
-        const obj: any = {};
-        message.transaction_id !== undefined && (obj.transaction_id = message.transaction_id);
-        message.email !== undefined && (obj.email = message.email);
-        message.phone !== undefined && (obj.phone = message.phone);
-        message.ip_address !== undefined && (obj.ip_address = message.ip_address);
-        message.unknown_user_field !== undefined && (obj.unknown_user_field = message.unknown_user_field);
-        return obj;
-    }
-}
-
-function isSet(value: any): boolean {
-    return value !== null && value !== undefined;
-}
+/* eslint-disable */
+/* istanbul ignore file */ 
+export interface AuditEvent {
+    event_id: string;
+    client_id: string;
+    timestamp: number;
+    timestamp_formatted: string;
+    event_name: string;
+    component_id: string;
+    user: IAuditEventUserMessage | undefined;
+    platform: unknown | undefined;
+    restricted: unknown | undefined;
+    extensions: unknown | undefined;
+    new_unknown_field: string;
+}
+
+export interface IAuditEventUserMessage {
+    transaction_id: string;
+    email: string;
+    phone: string;
+    ip_address: string;
+    session_id?: string;

```

Comment:
Why is this optional here?

Reply:
I believe they weren't required fields before and that is still the case now? It allows us to not need to provide them when building or test data if we so choose. I.e. in the first unit test we validate the minimum payload. We could arguably remove them now we have put them into the user object.

Reply:
They are optional, but so are the others here. Why are these the only ones for this file?